BGM ,Server list and autoupdate updates

sleeser opened the pull request:
Menu song change from song.ogg to song.mp3 and was moved to the sounds folder.
Added disadvantage and advantage bgms (L0/L1 >= 2)
Multiple bgms are now supported for duels, the game will pick a random one from the bgm folder at the start of the duel.
An option to randomize the bgms in a duel so when one is finished a new one will be picked or looping the current playing bgm.
Server list is no longer hard coded, a new file name "servers.conf" was added to the conifg folder and will be downloaded when its missing. (the update client function was updated so it doenst delete files anymore)
The auto updater now generates the SHA-1 of existing local files in the config and cdb folders and compare that to what is in the update git and then updates the files with different SHA hashes.
(No longer generates "SHAs.txt" file and the updates folder is no longer needed)
Fixed bugs related to the game creating empty conf files and trying to read them afterwards resulting in infinite loops.
Fixed a bug where no deck is selected when starting a duel resulting in immediate disconnections.

szefo09 commented on Assets/SibylSystem/Program.cs:
```diff
@@ -1,4 +1,1105 @@
+//    {
+//        PrintToChat(InterString.Get("非常抱歉，因为技术原因，此功能暂时无法使用。请关注官方网站获取更多消息。"));
+//    }
+//}
```
Why is all of this commented?

sleeser replied:
Leftovers i forgot to delete

szefo09 commented on Assets/SibylSystem/Program.cs:
```diff
@@ -429,6 +1531,86 @@ private void UpdateClient()
+                }
+                //foreach (string f in local)
+                //{
+                //    if (apiFromGit.FirstOrDefault(x => x.name == f) == null || f != apiFromGit.FirstOrDefault(x => x.name == f).name)
```
Why do you no longer want to remove unnecessary files?

sleeser replied:
Because it deletes config.conf because it isnt on the udpate git

Commit:
Update .gitignore for macOS development

Commit:
Update ocgcore

Commit:
Changes made by Unity to compile.

sleeser committed:
BGM Feature update

-- Advantage Music
-- Disadvantage Music
-- Random Music

sleeser committed:
Auto Update &&  Server.conf

-- Updates the updating system to compare git hashes and update files if the hashes don't match.
-- Moves the server picker list to server.conf so it is easier to update in the future.
